tests/spanner_dbapi/test_checksum.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Google LLC
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bsd
+
+import unittest
+
+from google.cloud.spanner_dbapi.checksum import (
+    _compare_checksums,
+    ResultsChecksum,
+)
+
+
+class Test_compare_checksums(unittest.TestCase):
+    def test_no_original_checksum(self):
+        self.assertIsNone(_compare_checksums(None, ResultsChecksum()))
```

Comment:
I would say that this should cause an error. Assume that the `_compare_checksums` function is called during a transaction retry. Then the following situations should cause the retry to be aborted because of differing results during the retry when compared to the initial attempt:
1. A query that did not return any results during the initial attempt does return results during the retry.
1. A query returns different results during the retry than the initial attempt.
1. A query returns an error during the retry, and did not return an error during the initial attempt (or vice-versa).

Reply:
This logic is no more actually needed. It was done for the original Spanner client, but here it's pointless.